src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -113,6 +122,10 @@ public NeuralQueryBuilder(StreamInput in) throws IOException {
         }
         this.k = in.readVInt();
```

Comment:
I am bit worried on this one now. As this Int can be null here, should we not add K as optional int if version is > 2.14

Reply:
Sure, I added conditional check for the K value if version is >= 2.14